modules/monitoring-setting-up-pod-topology-spread-constraints-for-prometheus.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: PROCEDURE
+[id="setting-up-pod-topology-spread-constraints-for-prometheus_{context}"]
+= Setting up pod topology spread constraints for Prometheus
+
+For core {product-title} platform monitoring, you can set up pod topology spread constraints for Prometheus to fine tune how pod replicas are scheduled across zones.
+Doing so helps ensure that Prometheus pods are highly available and run more efficiently because workloads are spread across different data centers or hierarchical infrastructure zones.
+
+You configure pod topology spread constraints for core platform monitoring in the `cluster-monitoring-config` config map.
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have created the `cluster-monitoring-config` `ConfigMap` object.
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` namespace:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add  values for the following settings under `data/config.yaml/prometheusK8s` and `data/config.yaml/alertmanagermain` to configure topology spread constraints:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      topologySpreadConstraints:
+      - maxSkew: 1 <1>
+        topologyKey: monitoring <2>
+        whenUnsatisfiable: DoNotSchedule <3>
+        labelSelector: 
+          matchLabels: <4>
+            prometheus: k8s
+----
+<1> Specify a numeric value for `maxSkew`, which defines the degree to which pods can be unevenly distributed. 
+This field is required, and the value must be greater than zero. 
+The value specified has a different effect depending on what value you specify for `whenUnsatisfiable`.
+<2> Specify a node label for `topologyKey`.
+This field is required.
+Nodes that match this label are considered to be in the same domain, and. 
+the pod scheduler tries to balance the number of pods in each domain.
+The value `non-exist` indicates that a node label does not exist and that therefore the pods can be scheduled to any node, whether or not the node has the `monitoring` label.
```

Comment:
I didn't find a reference to this `non-exist` value, do you have a pointer where I can read more info on it?

Reply:
This was mentioned in the text case in Polarion at https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-47393

Reply:
`non-exist` in the case just mean the topologyKey does not exist, you can replace to any value, that's does not mean you must set `non-exist`

Reply:
Ok. Thanks! I'll remove this.